pkg/provider/oracle/oracle.go

```diff
@@ -163,7 +164,16 @@ func (vms *VaultManagementService) NewClient(ctx context.Context, store esv1beta
 		configurationProvider common.ConfigurationProvider
 	)
 
-	if oracleSpec.Auth == nil {
+	if oracleSpec.PrincipalType == esv1beta1.WorkloadPrincipal {
+		// OCI SDK requires specific environment variables for workload identity.
+		if err := os.Setenv(auth.ResourcePrincipalVersionEnvVar, auth.ResourcePrincipalVersion2_2); err != nil {
```

Comment:
`os.Setenv` is not thread-safe. The controller can be configured to reconcile many resources concurrently. Is it only needed for the `auth.OkeWorkloadIdentityConfigurationProvider()` call below? If so, please please add a :lock: mutex lock for this section, otherwise we may need to add a lock for the provider (`NewClient()` acquires the lock, `Close()` releases it) .

Also: the `os.Setenv` should be cleaned up after use, otherwise it could interfere later on.

Reply:
Thanks for the suggestion, I've updated the PR.